### ISSUE TITLE
Auto-fill search field with transferred term

### ISSUE THREAD

**Issue by gaydenko** [itkach/aard2-android#59]
Hi! Let's look at the steps:

- invoke Aard from external application (I use AlReader)
- we have got article screen
- we want to refine search term
- we switch to search screen

Expected: search field is filled with the term was sent from external app to Aard.
But we get: search field is filled with old term used previous time we used the field.

**Comment by itkach** [comment]
Indeed. I remember struggling with getting Android task navigation to do what I wanted in all cases. There is navigation within application, navigation when lookup is initiated from external application and interaction between them. For example, user may initiate lookup from external app and then navigate away without closing dictionary task, and circle back to dictionary from the same external app, or do another lookup from another app, or launch Aard directly, or maybe it was already running. Tapping "home" (logo) button takes you all the way "back" to the top, except in case when lookup was launched externally it's not really going "back", just finds (or launches) dictionary's main task, which is unaware of the external lookup. It is broken in another way too - if you hit back button again it just closes Aard and does not take you back to your external app as might have hoped. This implementation is a compromise that seems to do reasonable things in main cases, but it's not great. Perhaps I'll give it another shot, but I'm not too optimistic, as Android task management didn't get any simpler and I didn't get any smarter.

**Comment by gaydenko** [comment]
> It is broken in another way too - if you hit back button again it just closes Aard and does not take you back to your external app as might have hoped.

Exactly! This is another issue I have met. It is still smaller as far as in most cases it results in just one or two extra tap(s).

> ... as Android task management didn't get any simpler and I didn't get any smarter.

I'm sure the last isn't true :) Good luck!

**Comment by gaydenko** [comment]
Another issue - impossible to search when book reader returns a baloon message a word was not found - move me to idea how to resolve all problems in one shot :) Let's open a screen with input field on book reader (or other application) request to translate/find! This way we

- eliminate "not found" problem
- permit to select an answer from different sources
- allow to immediately refine search context (say, remove plural form)

What do you think?
